sdv/metadata/multi_table.py

```diff
@@ -753,9 +755,12 @@ def _validate_missing_tables(self, data):
     def _validate_all_tables(self, data):
         """Validate every table of the data has a valid table/metadata pair."""
         errors = []
+        warning_dataframes = []
         for table_name, table_data in data.items():
+            table_sdtype_warnings = defaultdict(list)
             try:
-                self.tables[table_name].validate_data(table_data)
+                with warnings.catch_warnings(record=True):
```

Comment:
since we're catching the warnings, why not access them directly instead of adding them to a list?
something like
```python
try:
    with warnings.catch_warnings(record=True) as caught_warnings:
        ...

finally:
    for warning in caught_warnings:
        message = warning.message
        do_what_you_want_with_message()
```
Or was it just easier to add to a list?